emma-language/src/test/scala/org/emmalanguage/compiler/lang/core/DSCFSpec.scala

```diff
@@ -28,14 +28,14 @@ class DSCFSpec extends BaseCompilerSpec {
   val dscfPipeline: u.Expr[Any] => u.Tree =
     pipeline(typeCheck = true)(
       Core.anf,
-      tree => time(DSCF.transform(tree), "dscf"),
+      TreeTransform("DSCFSpec.dscfPipeline/DSCF.transform", tree => time(DSCF.transform(tree), "dscf")),
```

Comment:
It is still not clear to me why all these changes are needed. We don't gain anything here. Keeping the lines unchanged (or just changing the second parameter of the `time` from "dscf" to `DSCF.transform.name`) seems much cleaner. The implicit constructor should lift `tree => time(...)` lambdas to an "anonymous" `TreeTransform`.

Reply:
Agreed. I was sceptical about those too. I'll revert the changes and add an implicit.